### PR TITLE
feat: add Video component

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes:

Adds a `<Video />` component to replace manually adding HTML video element. Implementation and preview links can be found in this separate PR https://github.com/aws-amplify/docs/pull/7503 that converts all inline video to use this component.

**Usage**
```
<Video src="path/to/video.mp4" />

// Override width
<Video width="40%" src="path/to/video.mp4" />
```

- Adds consistent styling to video elements.
- Defaults videos to autoplay, muted, 100% width, playsinline, and loop. Authors can still pass these as attributes/props to override the defaults.
- Addresses [WCAG SC 2.2.2](https://www.w3.org/WAI/WCAG21/Understanding/pause-stop-hide.html) "Pause, Stop, Hide" by adding video controls. Additionally, video will only auto play if user has not checked prefers-reduced-motion in their operating system.
- Defaults video to `aria-hidden="true"`. When reviewing video usage, it's most often the case that we are showing a video that closely matches a list of instructions in text content. It's likely that if we add a text description for the video, it will be redundant and closely match text in the surrounding content and then create more noise for screen readers. In this scenario, it should be safe to hide the videos from screen readers.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by jacoblogan — DISMISSED** [review]

File: src/components/Video/Video.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const [prefersReducedMotion, setPrefersReducedMotion] =
+    useState<boolean>(true);
+
+  const videoRef = useRef<HTMLVideoElement>(null);
+
+  useEffect(() => {
+    const mediaQueryList = window.matchMedia(reducedMotionMediaQuery);
+
+    setPrefersReducedMotion(
+      !window.matchMedia(reducedMotionMediaQuery).matches
```
Comment (timngyn) [inline src/components/Video/Video.tsx:42]: nit: Could we reuse mediaQueryList from above for this line?
```suggestion
      !mediaQueryList.matches
```

File: src/components/Video/__tests__/Video.test.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }))
+  });
+}
+
+// Mock HTMLMediaElement.prototype.pause
+jest
+  .spyOn(window.HTMLMediaElement.prototype, 'pause')
+  .mockImplementation(() => {});
+
+const playSpy = jest
```
Comment (jacoblogan) [inline src/components/Video/__tests__/Video.test.tsx:27]: I don't think playSpy is ever being reset, making these tests (callCounts) order dependent.

File: src/components/Video/Video.tsx (hunk 1 above)
Comment (hbuchel) [reply src/components/Video/Video.tsx]: Thanks for the catch, fixed in latest

**Review by timngyn — APPROVED** [review]

File: src/components/Video/__tests__/Video.test.tsx (hunk 2 above)
Comment (hbuchel) [reply src/components/Video/__tests__/Video.test.tsx:27]: Ah thanks, I think you're right, and also explains why I was having trouble with a test earlier. In the latest I added a:
```
afterEach(() => {
    jest.clearAllMocks();
  }); 
```
And updated the tests to be a little more specific (one tests with the media query set to true, one now explicitly sets it to false in the test).

**Review by jacoblogan — APPROVED** [review]

**Pull request merged**